subtests/docker_cli/run_env/run_env.py

```diff
@@ -3,493 +3,123 @@
-        """\n"""
-        """conn.close()\n"""
-        """\n"""
-        """sys.exit(0)\n"""
-    )
-
-    def record_iptables(self, label):
-        name = ('%s_iptables_%s.txt'
-                % (self.__class__.__name__, label))
-        output = open(os.path.join(self.parent_subtest.resultsdir, name), 'w+')
-        output.write(utils.run('iptables -t filter -L -n -v').stdout)
-        output.write('\n\n')
-        output.write(utils.run('iptables -t nat -L -n -v').stdout)
-
-    def get_env(self, output):
-        """
-        Check output for token, return value as if parsed in python
-        """
-        for line in output.splitlines():
-            if line.startswith('ENVIRON = {'):
-                return ast.literal_eval(line[10:])
-        self.failif(True, "Fail to get env from output\n%s" % output)
-
-    def check_env(self, exp, variable, env, container, appendix):
-        """
-        Check that variable's value matches the exp regexp. Fails the test
-        when it's not.
-        :param exp: Expected value (string)
-        :param variable: Name of the env variable (string)
-        :param env: Environment (dictionary)
-        :param container: Name of the container (string)
-        :param appendix: Rest of the error string
-        """
-        self.failif(not re.match(exp, env.get(variable)), "%s=%s not found in"
-                    " %s env:\n%s%s" % (variable, exp, container, env,
-                                        appendix))
-
-    def init_container(self, prefix, subargs, cmd):
-        """
-        Prepares dkrcmd and stores the name in self.sub_stuff['containers']
-        :return: tuple(dkrcmd, name)
-        """
-        name = self.sub_stuff['dc'].get_unique_name(prefix)
-        subargs.append("--name")
-        subargs.append(name)
-        self.logdebug("Queuing container %s for removal", name)
-        self.sub_stuff['containers'].append(name)
-        fin = DockerImage.full_name_from_defaults(self.config)
-        subargs.append(fin)
-        subargs.append(cmd)
-        dkrcmd = InteractiveAsyncDockerCmd(self, 'run', subargs)
-        dkrcmd.verbose = True
-        dkrcmd.quiet = False
-        return dkrcmd, name
-
-    def initialize(self):
-        super(run_env_base, self).initialize()
-        # Prepare a container
-        config.none_if_empty(self.config)
-        self.sub_stuff['dc'] = DockerContainers(self)
-        self.sub_stuff['containers'] = []
-
-    def cleanup(self):
-        super(run_env_base, self).cleanup()
-        if self.config['remove_after_test']:
-            dc = DockerContainers(self)
-            dc.clean_all(self.sub_stuff.get("containers", []))
-
-
-class port_base(run_env_base):
-
-    """
-    Base class for port test.
-    """
+    pass
+
+
+class spam(SubSubtest):
+
+    # Allows quickly rejecting container env. supplied vars
+    KEYPFX = '_x'
+    VALPFX = 'x_'
+
+    @classmethod
+    def make_test_dict(cls, howmany, keyiter=None, valiter=None):
```

Comment:
This is confusing: we're never invoked with keyiter/valiter, so that's just dead code. And if we were ever called that way, the signature would be confusing: howmany would be ignored, making for a hard-to-understand call. Have I missed something? If I haven't, might it make sense to remove the keyiter/valiter code?

While I'm at it: what would you think of renaming to just test_dict() ? "make", to me, implies an action with a side effect. "make" is the sort of thing I expect to be invoked as an imperative:

    self.make_test_dict(...)

Without make, the invocation in line 112 becomes the slightly-more-readable (to me):

    self.sub_stuff['envd'] = self.test_dict(...)

...but that's just a minor nit.